### PR TITLE
Imported header plugin from bap-plugins

### PR DESCRIPTION
**Pull request by ivg**
The new name for the header is api. The pass utilizes
the new `autoload` feature, and applies known APIs on
the program. APIs are bundled with the plugin itself.
New API's can be added by providing a C header file.

### CODE REVIEW

**Comment by rvantonder** [comment]
It might be a little hard for someone outside of BAP to figure out what api means, because it's such a general term. Maybe scope it for function or subroutine APIs? But I don't have a better suggestion right now. Otherwise, happy to see this in main BAP :+1:

File: plugins/api/Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+header.plugin : header.ml
```
Comment (rvantonder) [inline plugins/api/Makefile]: Doesn't look like there's a `header.ml` file in this commit.

Comment (ivg) [inline plugins/api/Makefile]: ah, forget to remove the Makefile :)

**Comment by ivg** [comment]
The name is chosen from the perspective of a user, how it will invoke it: 
```
bap --api-add odbx.h 
```
Other than this, there is no need to call it, it will work out of box.

**Pull request merged**